examples/FungibleToken/src/lib.rs

```diff
@@ -0,0 +1,164 @@
+use primitive_types::H256;
+use scale_info::TypeInfo;
+
+#[derive(Debug)]
+struct FungibleToken {
+    name: String,
+    symbol: String,
+    total_supply: u128,
+    balances: BTreeMap<ProgramId, u128>,
+}
+
+static mut FUNGIBLE_TOKEN: FungibleToken = FungibleToken {
+    name: String::new(),
+    symbol: String::new(),
+    total_supply: 0u128,
+    balances: BTreeMap::new(),
+};
+
+impl FungibleToken {
+    fn set_name(&mut self, name: String) {
+        self.name = name;
+    }
+    fn name(&self) -> String {
+        self.name.clone()
+    }
+    fn set_symbol(&mut self, symbol: String) {
+        self.symbol = symbol;
+    }
+    fn symbol(&self) -> String {
+        self.symbol.clone()
+    }
+    fn total_supply(&self) -> u128 {
+        self.total_supply
+    }
+    // TODO: Use checked math and return error if Overflow.
+    fn increase_total_supply(&mut self, amount: u128) {
+        self.total_supply += amount;
+    }
+    // TODO: Use checked math and return error if Underflow.
+    fn decrease_total_supply(&mut self, amount: u128) {
+        self.total_supply -= amount;
+    }
+    fn set_balance(&mut self, account: &ProgramId, amount: u128) {
+        self.balances.insert(*account, amount);
+    }
+    fn get_balance(&self, account: &ProgramId) -> u128 {
+        *self.balances.get(account).ok_or(0u128).unwrap()
+    }
+    fn mint(&mut self, account: &ProgramId, amount: u128) {
+        unsafe {
+            FUNGIBLE_TOKEN.increase_total_supply(amount);
+            let old_balance = FUNGIBLE_TOKEN.get_balance(account);
+            // TODO: safe math for overflow detection.
+            FUNGIBLE_TOKEN.set_balance(account, old_balance + amount);
+        }
+    }
+    fn burn(&mut self, account: &ProgramId, amount: u128) {
+        unsafe {
+            FUNGIBLE_TOKEN.decrease_total_supply(amount);
+            let old_balance = FUNGIBLE_TOKEN.get_balance(account);
+            // TODO: safe math for underflow detection.
+            FUNGIBLE_TOKEN.set_balance(account, old_balance - amount);
+        }
+    }
+}
+
+#[derive(Debug, Decode, TypeInfo)]
+struct InitConfig {
+    name: String,
+    symbol: String,
+}
+
+#[derive(Debug, Decode, TypeInfo)]
+struct MintInput {
+    account: H256,
+    amount: u128,
+}
+
+#[derive(Debug, Decode, TypeInfo)]
+struct BurnInput {
+    account: H256,
+    amount: u128,
+}
+
+#[derive(Debug, Decode, TypeInfo)]
+enum Action {
+    Mint(MintInput),
+    Burn(BurnInput),
+}
+
+#[derive(Debug, Encode, TypeInfo)]
+struct TransferData {
+    from: ProgramId,
+    to: ProgramId,
+    amount: u128,
+}
+
+#[derive(Debug, Encode, TypeInfo)]
+enum Event {
+    Transfer(TransferData),
+}
+
+gstd::metadata! {
+    title: "FungibleToken",
+        init:
+            input : InitConfig,
+            output : (),
+        handle:
+            input : Action,
+            output : Event,
+}
+
+#[no_mangle]
+pub unsafe extern "C" fn handle() {
+    let action: Action = msg::load().unwrap();
```

Comment:
```suggestion
    let action: Action = msg::load().expect("Unable to decode Action");
```

It would be better to know panic's cause